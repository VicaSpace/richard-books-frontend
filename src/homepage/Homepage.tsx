import React from 'react'
import './Homepage.css'

const Homepage: React.FC<{ showTitle: boolean }> = ({ showTitle }) => {
  return (
    <div className="homepage-title">
      {showTitle && <p data-testid="title-test">Hello</p>}
      <p>Homepage</p>
    </div>
  )
}

export default Homepage
