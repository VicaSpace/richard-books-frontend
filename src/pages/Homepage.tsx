import axios from 'axios'
import React, { useEffect, useState } from 'react'
import BookHorizontalSlider from '../components/bookHorizontalSlider/BookHorizontalSlider'
import NavBar from '../utils/sharedComponents/navBar/NavBar'
import HeadingSection from '../components/headingSection/HeadingSection'
import PopularAuthors from '../components/popularAuthors/PopularAuthors'
import Premium from '../components/premium/Premium'

const HomePage: React.FC<{}> = () => {
  const [books, setBooks] = useState([])
  const [authors, setAuthors] = useState([])

  useEffect(() => {
    axios
      .get('http://localhost:3001/api/books')
      .then((res) => {
        setBooks(res.data)
      })
      .catch((err) => console.log(err))
    axios
      .get('http://localhost:3001/api/authors')
      .then((res) => {
        setAuthors(res.data)
      })
      .catch((err) => console.log(err))
  }, [])

  return (
    <div className='homepage-container'>
      <HeadingSection />
      <BookHorizontalSlider books={books} />
      <Premium />
      <PopularAuthors authors={authors} />
      <NavBar />
    </div>
  )
}

export default HomePage
