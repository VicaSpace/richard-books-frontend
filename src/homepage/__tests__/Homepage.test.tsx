import React from 'react'
import Homepage from '../Homepage'
import { render, screen } from '@testing-library/react'

describe('snapshot testing homepage', () => {
  it('should render the homepage correctly', () => {
    const { container } = render(<Homepage showTitle />)
    expect(container).toMatchSnapshot()
  })
})

describe('title exist', () => {
  it('should exists', () => {
    render(<Homepage showTitle />)
    const title = screen.getByTestId('title-test')
    expect(title).toBeInTheDocument()
  })

  it('shoud not exists', () => {
    render(<Homepage showTitle={false} />)
    const title = screen.queryByTestId('title-test')
    expect(title).not.toBeInTheDocument()
  })
})
