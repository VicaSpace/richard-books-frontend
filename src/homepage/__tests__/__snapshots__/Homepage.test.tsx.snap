// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing homepage should render the homepage correctly 1`] = `
<div>
  <div
    class="homepage-title"
  >
    <p
      data-testid="title-test"
    >
      Hello
    </p>
    <p>
      Homepage
    </p>
  </div>
</div>
`;
